src/resources/transactions/firstTransactionTimestampQuery.ts

```diff
@@ -0,0 +1,104 @@
+import { useQuery } from '@tanstack/react-query';
+import PQueue from 'p-queue/dist';
+
+import { fetchENSAddress } from '@/hooks/useENSAddress';
+import {
+  createQueryKey,
+  queryClient,
+  QueryConfig,
+  QueryFunctionArgs,
+  QueryFunctionResult,
+} from '@/react-query';
+import { getFirstTransactionTimestamp } from '@/utils/ethereumUtils';
+
+// ///////////////////////////////////////////////
+// Query Types
+
+export type FirstTransactionTimestampArgs = {
+  addressOrName?: string;
+};
+
+// ///////////////////////////////////////////////
+// Query Key
+
+export const firstTransactionTimestampQueryKey = ({
+  addressOrName,
+}: FirstTransactionTimestampArgs) =>
+  createQueryKey(
+    'firstTransactionTimestamp',
+    { addressOrName },
+    { persisterVersion: 1 }
+  );
+
+export type FirstTransactionTimestampQueryKey = ReturnType<
+  typeof firstTransactionTimestampQueryKey
+>;
+
+// ///////////////////////////////////////////////
+// Query Function
+
+const queue = new PQueue({ interval: 1000, intervalCap: 5 });
```

Comment:
Unrelated to this PR: I wonder if we'll still need this to rate limit these requests 🤔

Reply:
Good question! I don't think so.